src/test/java/com/dropbox/core/v2/teamlog/GetTeamEventsResultTest.java

```diff
@@ -0,0 +1,74 @@
+package com.dropbox.core.v2.teamlog;
+
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertTrue;
+
+public class GetTeamEventsResultTest {
+
+
+    /**
+     * Tests a known type that has extra fields
+     *
+     * Happy path for known objects
+     */
+    @Test
+    public void known_with_body() throws IOException {
+        GetTeamEventsResult result = readGetTeamEventsResultFromFile("known_with_body.json");
+        TeamEvent event = result.events.get(0);
+        assertEquals(EventDetails.Tag.FILE_LOCKING_LOCK_STATUS_CHANGED_DETAILS, event.details.tag());
+    }
+
+    /**
+     * Tests a known type that should have extra fields, but does not, so should throw exception
+     *
+     * Sad path for known objects without required fields
+     */
+    @Test
+    public void known_without_body() {
+        try {
+            GetTeamEventsResult result = readGetTeamEventsResultFromFile("known_without_body.json");
+            assertEquals(EventDetails.Tag.OTHER, result.events.get(0).details.tag());
```

Comment:
Shouldn't this never run because `readGetTeamEventsResultFromFile` should throw? A bit confusing since theres an assertion in both the try and catch blocks

Reply:
Thanks, removed.  Copy/paste artifact.